content/en/users/tutorials/jupyter-datahub-virtual-machine/_index.md

```diff
@@ -0,0 +1,119 @@
+that runs a browser accessible jupyter server with DataHub spaces mounted. This
+setup can be used in the EGI Federation or in any other provider which
+synchronise images from AppDB but is not part of the federation.
+
+## Requirements
+
+This tutorial assumes you have:
+
+- A valid EGI account: learn to can create one in
+  [Check-in](../../aai/check-in/signup).
+- Access to a cloud provider where the
+  [Jupyter DataHub VM is available](https://appdb.egi.eu/store/vappliance/jupyter.datahub.vm).
+  Alternatively, this VM can be run on your computer using a virtualization tool
+  like VirtualBox.
+
+## Create a VM with Jupyter and DataHub
+
+### Step 1: Start your VM
+
+Start your VM on your cloud provider or virtualization tool. You can check
+[the tutorial on how to start a VM](../create-your-first-virtual-machine) to
+learn how to start a VM at EGI's Federated Cloud infrastructure.
+
+This VM does not contain any default credentials, in order to access it you need
+a ssh key. Check
+[this FAQ entry](../../compute/cloud-compute/faq/#how-can-i-inject-my-public-ssh-key-into-the-machine)
+for more information. If you are starting this VM on VirtualBox, you will need
+to pass some valid context for cloud-init, see
+[here for how to prepare it](https://superuser.com/a/853957)
+
+The VM image is ready to listen on port `22` for ssh access and port `80` for
+accessing the notebooks interface. Make sure your have those ports open on your
+security groups, otherwise you will not be able to reach the Jupyter notebooks.
+
+Once your instance is ready,
+[assign it a public IP](../../compute/cloud-compute/faq/#how-can-i-assign-a-public-ip-to-my-vm)
+so you can reach it from your computer.
+
+### Step 2: Setup DataHub
+
+Log into [EGI's DataHub](https://datahub.egi.eu/) and
+[create a token](../../data/management/datahub/clients/#generating-tokens-for-using-oneclient-or-apis)
+for mounting your data in the VM.
+
+You will also need the IP or address of your closest provider for the spaces you
+are interested in accessing. This information is easily obtainable via
+[DataHub's web interface](../../data/management/datahub/clients/#using-the-web-interface)
+
+SSH to your VM as user `ubuntu`:
+
+```shell
+$ ssh ubuntu@<your VM's IP>
+```
+
+And edit the `mount.sh` script by adding the `ONECLIENT_ACCESS_TOKEN` and
+`ONECLIENT_PROVIDER_HOST` to get access to DataHub:
+
+```shell
+#!/bin/bash
+
+# Personal token obtained from https://datahub.eu
+ONECLIENT_ACCESS_TOKEN="<your DataHub token>"
+# Closest oneprovider supporting your spaces
+ONECLIENT_PROVIDER_HOST="plg-cyfronet-01.datahub.egi.eu"
+
+oneclient -H "$ONECLIENT_PROVIDER_HOST" -t "$ONECLIENT_ACCESS_TOKEN" /mnt/datahub
+```
+
+Now, mount your spaces by invoking the script:
+
+```shell
+$ mount.sh
+Connecting to provider 'plg-cyfronet-01.datahub.egi.eu:443' using session ID: '4173323292216088977'...
+Getting configuration...
+Oneclient has been successfully mounted in '/mnt/datahub'.
```

Comment:
`/mnt/datahub` has been configured for the `ubuntu` user.
I have used IM, which creates the `cloudadm` user and I get:
```
fusermount: user has no write access to mountpoint /mnt/datahub
```
Currently the manual solution is:
```
sudo chown cloudadm.cloudadm -R /mnt/datahub/
```

Reply:
I think I will just use jovyan as user for everything
Was trying to avoid root access as much as possible

Reply:
Mounting now done as `jovyan`, problem should be gone